_posts/2018-02-06-unrealiable-network.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: post
+title: Quick Tips for App Developers on Surviving with Unrealiable Network
+author: jarzka
+excerpt: Methods that you, application developer, can do to make your app feel more stable even if the underlying network is unreliable.
+tags:
+- clojure
+- clojurescript
+- network programming
+---
+
+You cannot always choose where your application will be hosted. Sometimes the network infrastructure can be unrealiable, meaning that network requests can sometimes take a long time to complete and requests will fail at random rates. Even if these problems could not be prevented from happening, luckily we, application developers, have some tricks we can use to make our applications *feel* more stable in unrealiable network. And even if your network infrastucture is solid, it is not guaranteed that occasional communication failures won't happen, as illustrated in [Two Generals Problem](https://en.wikipedia.org/wiki/Two_Generals%27_Problem). Thus, these quick tips, targeted mainly for single page web application and mobile application developers, will probably help your application behave more stable and quickly for your end users.
```

Comment:
> requests can sometimes take

requests can at times [poistaa sometimes-sanan toiston]

> luckily we, application developers

luckily we, as application developers

> communication failures won't happen

communication failures will not happen [tyyliseikka, mutta yleensä asiatekstissä ei ole tapana käyttää kontraktioita]

> targeted mainly for

targeted mainly *at*

> will probably help your application behave more stable and quickly for your end users.

will hopefully help your application appear more stable and responsive [toivottavasti ei sekoitu leiskan responsiivisuuteen] to your end users.

Reply:
Great suggestions!